### PR TITLE
KPIV-17: add more join conference options to discovery req

### PR DESCRIPTION
**Pull request by icehess**
This updates conference call command to allow more options when a member joins the conference without needing to update code to add new options.

Also updating discovery req to accept more headers so applications send more conference override settings.

Adds accessories for endconf conference flags.

### CODE REVIEW

**Comment by icehess** [comment]
- Master PRs:
    - https://github.com/2600hz/kazoo-core/pull/839
    - https://github.com/2600hz/kazoo-crossbar/pull/574
    - https://github.com/2600hz/kazoo-conference/pull/50
    - https://github.com/2600hz/kazoo-ecallmgr/pull/246
- 5.4 PRs:
    - https://github.com/2600hz/kazoo-core/pull/924
    - https://github.com/2600hz/kazoo-crossbar/pull/595
    - https://github.com/2600hz/kazoo-conference/pull/51
    - https://github.com/2600hz/kazoo-ecallmgr/pull/253

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 1]
```diff
@@ -9,12 +9,31 @@
...
+        },
         "Conference-ID": {
+            "minLength": 1,
             "type": "string"
         },
         "Conference-Name": {
+            "minLength": 1,
+            "type": "string"
+        },
+        "Controller-Queue": {
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: what's the reasoning for adding `Controller-Queue`  ?

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 2]
```diff
@@ -9,12 +9,31 @@
         "Conference-Doc": {
             "type": "string"
         },
+        "Conference-Focus": {
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: how is `Conference-Focus` used ?

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 3]
```diff
@@ -9,12 +9,31 @@
...
             "type": "string"
         },
         "Conference-Name": {
+            "minLength": 1,
+            "type": "string"
+        },
+        "Controller-Queue": {
             "type": "string"
         },
+        "Controls": {
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: how is `Controls` used ?

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 4]
```diff
@@ -9,12 +9,31 @@
...
+            "type": "object"
+        },
+        "Domain": {
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: how is `Domain` used ? afaics, `Domain` comes from account

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 5]
```diff
@@ -27,19 +46,84 @@
             ],
             "type": "string"
         },
+        "Language": {
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: how is `Language` used ?

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 6]
```diff
@@ -27,19 +46,84 @@
...
+            "minLength": 1,
+            "type": "string"
+        },
+        "Last-Member-EndConf": {
+            "type": "boolean"
+        },
+        "Last-Member-Endconf": {
+            "type": "string"
+        },
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: duplicated

File: kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json [hunk 7]
```diff
@@ -27,19 +46,84 @@
...
+            },
+            "type": "string"
+        },
         "Moderator": {
             "type": "string"
         },
+        "Moderator-Controls": {
+            "minLength": 1,
+            "type": "string"
+        },
```
Comment (lazedo) [inline kazoo_amqp/priv/couchdb/schemas/kapi.conference.discovery_req.json]: how is `Moderator-Controls` used ?

**Review by lazedo — COMMENTED** [review]
need some context for all these unrelated changes

**Comment by icehess** [comment]
Hi, I'm going to move some of those extra  changes to other KPIV-17 prs. But the idea was to be able to override more conference parameters on discovery. And you right, some of those headers like domain and controller-queue were added by mistake.

**Review by lazedo — COMMENTED** [review]
`endconf` is a flag and shouldn't be part of the conference record

File: kazoo_call/src/kapps_call_command.erl [hunk 8]
```diff
@@ -2372,13 +2372,18 @@ conference(ConfId, Mute, Deaf, Moderator, ProfileName, Reinvite, Call) ->
...
-              ,{<<"Video-Mute">>, VideoMute}
+              ,{<<"Join-Options">>
+               ,kz_json:from_list(
+                  [{<<"Mute">>, Mute}
+                  ,{<<"Deaf">>, Deaf}
+                  ,{<<"Moderator">>, Moderator}
+                  ,{<<"Video-Mute">>, VideoMute}
+                  ]
+                 )
+               }
```
Comment (lazedo) [inline kazoo_call/src/kapps_call_command.erl:2386]: what about `End-Conference-On-Leave` ?

**Pull request merged**